### PR TITLE
Support z-index in geckolib

### PR DESCRIPTION
**Pull request by mbrubeck**
r? @bholley

cc @heycam

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11176)

### CODE REVIEW

File: ports/geckolib/properties.mako.rs [hunk 1]
```diff
@@ -497,13 +497,31 @@ fn static_assert() {
...
+
+    fn set_z_index(&mut self, v: longhands::z_index::computed_value::T) {
+        use gecko_bindings::structs::nsStyleUnit;
+        match v {
+            longhands::z_index::computed_value::T::Auto => {
+                self.gecko.mZIndex.mUnit = nsStyleUnit::eStyleUnit_Auto;
+                unsafe { *self.gecko.mZIndex.mValue.mInt.as_mut() = 0; }
+            }
+            longhands::z_index::computed_value::T::Number(n) => {
+                self.gecko.mZIndex.mUnit = nsStyleUnit::eStyleUnit_None;
```
Comment (bholley) [inline ports/geckolib/properties.mako.rs]: Don't we want eStyleUnit_Integer here?

File: ports/geckolib/properties.mako.rs [hunk 2]
```diff
@@ -497,13 +497,31 @@ fn static_assert() {
...
     % for side in SIDES:
     <% impl_split_style_coord("%s" % side.ident,
                               "mOffset.mUnits[%s]" % side.index,
                               "mOffset.mValues[%s]" % side.index) %>
     % endfor
+
+    fn set_z_index(&mut self, v: longhands::z_index::computed_value::T) {
+        use gecko_bindings::structs::nsStyleUnit;
+        match v {
+            longhands::z_index::computed_value::T::Auto => {
```
Comment (bholley) [inline ports/geckolib/properties.mako.rs]: seems like it would be cleaner to |use| this type.

**Comment by bholley** [comment]
r=me with those fixes. @heycam should give it a quick lookover, but no need to block landing on that.

**Comment by mbrubeck** [comment]
@bors-servo retry

* #8815

**Comment by heycam** [comment]
LGTM. When we come to implement column-count in geckolib, we might want to factor out the handling of z-index and column-count into a new IntegerOrAutoOrNone type to avoid some duplication in components/style/properties/longhand/ and also so that we can handle this named value type in ports/geckolib/properties.mako.rs (by adding a ToGeckoStyleCoord impl for it).

**Pull request merged**